lib/screens_web/views/v2/audio/reconstructed_alert_fullscreen_view.ex

```diff
@@ -0,0 +1,206 @@
+defmodule ScreensWeb.V2.Audio.ReconstructedAlertFullscreenView do
+  use ScreensWeb, :view
+
+  alias Screens.Alerts.Alert
+  alias Screens.Util
+
+  def render("_widget.ssml", alert) do
+    ~E|<p><%= render_banner(alert) %><%= render_alert(alert) %></p>|
+  end
+
+  def render_banner(%{
+        unaffected_routes: _unaffected_routes
+      }),
+      do: nil
+
+  def render_banner(%{
+        region: _region,
+        routes: routes
+      }) do
+    if routes |> hd() |> Map.has_key?(:headsign) do
+      destinations = Enum.map(routes, fn route -> route.headsign end)
+
+      if length(destinations) < 3 do
+        ~E|Attention, riders to <%= Util.format_name_list_to_string_audio(destinations) %>. |
+      else
+        ~E|Attention, riders. |
+      end
+    else
+      ~E|Attention, <%= hd(routes).route_id %> riders. |
```

Comment:
So is it a rule that if the first map in `routes` does not have a `:headsign` key, then we only care about its `:route_id` key instead? And don't care about the rest of the list?

In general, this code seems to make implicit assertions about the shape of the list and its elements that are not made very clear—this makes the code difficult to follow since I can't look at a typespec or doc comment to understand what the data looks like. I have to look back at the reconstructed alert widget code and walk through the logic to try and parse out the different possible shapes of these maps.

Reply:
Woof, yeah, sorry this was so convoluted. I don't see a great way of simplifying, but I did add comments to explain. Was the rest of the file commented well enough to follow?